src/main/java/programminglife/model/Graph.java

```diff
@@ -79,79 +84,52 @@ void parseLink(Scanner sc) {
         destinationNode.addParent(sourceNode);
     }
 
-    public static Graph parse(String file) throws FileNotFoundException {
+    public static Graph parse(String file) throws FileNotFoundException, UnknownTypeException {
         return parse(file, PARSE_LINE_VERBOSE_DEFAULT);
     }
 
-    public static Graph parse(String file, boolean verbose) throws FileNotFoundException {
-        return parse(file, verbose, PARSE_LINE_LOG_INTERVAL_DEFAULT);
-    }
-
-    public static Graph parse(String file, boolean verbose, int lineInterval) throws FileNotFoundException {
+    public static Graph parse(String file, boolean verbose) throws FileNotFoundException, UnknownTypeException {
         if (verbose) {
             System.out.println(String.format("Parsing file %s", file));
         }
 
-        Scanner lineScanner = new Scanner(new File(file));
+        BufferedReader reader = new BufferedReader(new FileReader(file));
         Graph graph = new Graph(null);
-        boolean logCurrentLine = false;
-
-        int numLinesRead = 0;
```

Comment:
This is not possible using a lambda with the current method. We can look into using a counter/statistics/progress object.